WebHostLib/customserver.py

```diff
@@ -277,12 +277,15 @@ async def start_room(room_id):
                     ctx.logger.exception("Could not determine port. Likely hosting failure.")
                 with db_session:
                     ctx.auto_shutdown = Room.get(id=room_id).timeout
+                if ctx.saving:
+                    setattr(asyncio.current_task(), "save", lambda: ctx._save(True))
```

Comment:
I had no idea one could do this, and admittedly it feels somewhat cursed.

Reply:
It really does. I think the alternative would be keeping yet another list of active rooms. Not sure if that would be better, but I feel like it would be another source of memory leaks.